R/model.R

```diff
@@ -934,7 +934,13 @@ as_generator.tensorflow.python.data.ops.dataset_ops.DatasetV2 <-
   as_generator.tensorflow.python.data.ops.dataset_ops.Dataset
 
 as_generator.function <- function(x) {
-  reticulate::py_iterator(function() keras_array(x()))
+  python_path <- system.file("python", package = "keras")
+  tools <- reticulate::import_from_path("kerastools", path = python_path)
+  iter <- reticulate::py_iterator(function() {
+    elem <- x()
```

Comment:
We lost the call to `keras_array()` here. Let's bring it back.

Reply:
TensorFlow 1.3 assumes a tuple is provided, so we actually need this change. However, we did not verified against 1.12, which just did and tests are also passing locally, adding Travis now.

You happen to know when this might break? I'll add it back for `tf < 1.12`.

Reply:
You should be able to do this:

```r
elem <- keras_array(x())
reticulate::tuple(elem[[1]], elem[[2]])
```

Or perhaps: